detox/local-cli/detox-run-server.js

```diff
@@ -2,14 +2,11 @@
 
 const program = require('commander');
 const cp = require('child_process');
-const fs = require('fs');
 const path = require('path');
 
 program.parse(process.argv);
 
-if (fs.existsSync(path.join(process.cwd(), 'node_modules/.bin/detox-server'))) {
```

Comment:
npm and yarn install detox-server at different locations.
it might be either it `node_modules` or in `node_modules/detox/node_modules`

I am not familiar with `require.resolve('detox-server/package.json');` trick. Will it cover both cases ?

Reply:
Yep, it's the standard hack to find the location of an NPM package the same way node does, since every NPM package has a `package.json`. Remember:
- `require / require.resolve('foo/bar')` looks up `foo` in node_modules, then adds `bar` to that path
- without the `.../bar` it would use the `main` field, which is likely to be in a nested subdir
- `package.json` always exists.